algorithms/src/msm/variable_base/standard.rs

```diff
@@ -75,39 +75,354 @@ fn process_window<G: AffineCurve>(
         res += running_sum;
     }
 
-    res
+    (res, window_size)
+}
+
+fn batched_window<G: AffineCurve>(
```

Comment:
maybe make these as inherent methods of `MSMStrategy`? So you can call, e.g., `strategy.window(...)`. That would simplify the code below.